app/controllers/api/queries_controller.rb

```diff
@@ -62,11 +62,24 @@ def show
     def create
       options = {}
       options[:filters] = build_filter
-
+      
       authorize_providers
+      end_date = params[:effective_date]
+      start_date = params[:effective_start_date]
+
+      end_date = Time.at(end_date.to_i) if end_date.is_a?(String)
```

Comment:
OK. I'm not sure how a Time object could end up there, but if it does, I'll take your word for it.